src/plugins/data_source/server/legacy/http_aws_es/connector.js

```diff
@@ -94,7 +94,12 @@ class HttpAmazonESConnector extends HttpConnector {
   createRequest(params, reqParams) {
     const [pathname = '/', queryStr = ''] = (reqParams.path || '').split('?', 2);
 
-    const queryParams = queryStr ? queryString.parse(queryStr) : undefined;
+    const queryParams = {};
+    if (queryStr) {
+      for (const [key, value] of new URLSearchParams(queryStr)) {
+        queryParams[key] = value ?? '';
+      }
```

Comment:
```suggestion
      const params = new URLSearchParams(queryStr);
      for (const [key, value] of params) {
        queryParams[key] = value ?? '';
      }
```

To avoid recreating of ```URLSearchParams```

Reply:
Then the for...of iterates over the entries (key-value pairs) inside that already-created URLSearchParams object.

So only one URLSearchParams object is created for the entire loop, not in each iteration.